### PR TITLE
Generic Info REMOVE_DELAY to remove a job after a configurable time

### PR DESCRIPTION
**Pull request by fviale**
- Add functional test for housekeeping

- Also, fix issue with ServerJobAndTaskLogs while removing precious logs (access to SchedulerSpacesSupport from a non-active object context).
Fix minor issues appearing in the code

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 1]
```diff
@@ -949,14 +954,28 @@ static RuntimeException launderThrowable(Throwable t) {
...
             if (SchedulingService.SCHEDULER_AUTO_REMOVED_JOB_DELAY > 0) {
-                long timeToRemove = System.currentTimeMillis() + SchedulingService.SCHEDULER_AUTO_REMOVED_JOB_DELAY;
+                removeDelay = SchedulingService.SCHEDULER_AUTO_REMOVED_JOB_DELAY;
+            }
+            if (jobGenericInfo != null && jobGenericInfo.containsKey(GENERIC_INFO_REMOVAL_DELAY)) {
+                try {
+                    removeDelay = Math.min(removeDelay,
+                                           Tools.formatDate(jobGenericInfo.get(GENERIC_INFO_REMOVAL_DELAY)));
+                } catch (Exception e) {
+                    logger.error("Error when parsing generic information " + GENERIC_INFO_REMOVAL_DELAY + " for job " +
```
Comment (mcompastie) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java:971]: I would advice to remind in the log the value of the default removal delay to ease the log exploitation in debugging session.

Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java]: It is not necessary as I now print in INFO the exact data when the job is removed

**Comment by aminelouati** [comment]
May be you can add a small paragraph in the doc about REMOVE_DELAY and its use

**Comment by fviale** [comment]
@aminelouati sure, I will add it

**Pull request merged**